src/libDirectoryService/DirectoryService.cpp

```diff
@@ -677,6 +710,174 @@ bool DirectoryService::ToBlockMessage([[gnu::unused]] unsigned char ins_byte) {
+          m_mediator.m_selfPeer, get_time_as_int(), m_mediator.m_selfKey)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::SetDSLookupNewDSGuardNetworkInfo failed.");
+    return false;
+  }
+
+  // Send to all lookups
+  m_mediator.m_lookup->SendMessageToLookupNodesSerial(
+      updatedsguardidentitymessage);
+
+  {
+    // Gossip to all DS committee
+    lock_guard<mutex> lock(m_mediator.m_mutexDSCommittee);
+    vector<Peer> peerInfo;
+
+    for (auto const& i : *m_mediator.m_DSCommittee) {
+      if (i.second.m_listenPortHost != 0) {
+        peerInfo.push_back(i.second);
+      }
+    }
+
+    if (BROADCAST_GOSSIP_MODE) {
+      P2PComm::GetInstance().SpreadRumor(updatedsguardidentitymessage);
+    } else {
+      P2PComm::GetInstance().SendMessage(peerInfo,
+                                         updatedsguardidentitymessage);
+    }
+  }
+  m_awaitingToSubmitNetworkInfoUpdate = false;
+
+  return true;
+}
+
+bool DirectoryService::ProcessNewDSGuardNetworkInfo(
+    const std::vector<unsigned char>& message, unsigned int offset,
+    [[gnu::unused]] const Peer& from) {
+  LOG_MARKER();
+
+  if (!GUARD_MODE) {
+    LOG_GENERAL(WARNING,
+                "Not in guard mode. Unable to update ds guard network info.");
+    return false;
+  }
+
+  uint64_t dsEpochNumber;
+  Peer dsGuardNewNetworkInfo;
+  uint64_t timestamp;
+  PubKey dsGuardPubkey;
+
+  if (!Messenger::GetDSLookupNewDSGuardNetworkInfo(
+          message, offset, dsEpochNumber, dsGuardNewNetworkInfo, timestamp,
+          dsGuardPubkey)) {
+    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Messenger::GetDSLookupNewDSGuardNetworkInfo failed.");
+    return false;
+  }
+
+  if (m_mediator.m_selfKey.second == dsGuardPubkey) {
+    LOG_GENERAL(INFO, "Node to be update is current node. No update needed.");
+    return false;
+  }
+
+  uint64_t currentDSEpochNumber =
+      m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() + 1;
+  uint64_t loCurrentDSEpochNumber = currentDSEpochNumber - 1;
+  uint64_t hiCurrentDSEpochNumber = currentDSEpochNumber + 1;
+
+  if (!(dsEpochNumber <= hiCurrentDSEpochNumber + 1 &&
+        dsEpochNumber >= loCurrentDSEpochNumber - 1)) {
+    LOG_GENERAL(WARNING,
+                "Update of ds guard network info failure due to not within "
+                "range of expected "
+                "ds epoch loCurrentDSEpochNumber: "
+                    << loCurrentDSEpochNumber
+                    << " hiCurrentDSEpochNumber: " << hiCurrentDSEpochNumber
+                    << " dsEpochNumber: " << dsEpochNumber);
+    return false;
+  }
+
+  // Check timestamp
+  // Allowed only 5 mins. Else consider it obselete.
+  uint64_t allowableTimeForDSNodeNetworkInfoUpdate = 300;
+  if (!VerifyTimestamp(timestamp, allowableTimeForDSNodeNetworkInfoUpdate)) {
+    LOG_GENERAL(WARNING, timestamp << "exceeded "
```

Comment:
As we already have a similar error message inside `VerifyTimestamp`, maybe we don't need this duplicate warning?